src/components/UpdateSubscription/components/FreeExtension.tsx

```diff
@@ -0,0 +1,199 @@
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useAppDispatch, useAppSelector } from 'redux/store';
+import { hidePopup } from 'redux/popupSlice';
+import {
+  fetchApplyRetentionAction,
+  selectRetentionActions
+} from 'redux/retentionActionsSlice';
+import Button from 'components/Button';
+import Loader from 'components/Loader';
+import {
+  ContentStyled,
+  TitleStyled,
+  TextStyled,
+  ButtonWrapperStyled
+} from 'components/InnerPopupWrapper/InnerPopupWrapperStyled';
+import { LinkStyled } from 'components/ThankYouPage/ThankYouPageStyled';
+import {
+  ImageStyled,
+  ImageWrapper
+} from 'components/SwitchPlanPopup/SwitchPlanPopupStyled';
+import { ReactComponent as Close } from 'assets/images/errors/close.svg';
+import checkmarkIcon from 'assets/images/checkmarkBase';
+
+import {
+  FreeExtensionWrapperStyled,
+  TextWrapperStyled,
+  FreeExtensionCardStyled,
+  AcceptButtonWrapperStyled,
+  FreeExtensionCardPeriodStyled
+} from './FreeExtension.styled';
+
+type FreeExtensionProps = {
+  handleUnsubscribe: () => void;
+};
+
+const FreeExtension = ({ handleUnsubscribe }: FreeExtensionProps) => {
+  const { t } = useTranslation();
+  const dispatch = useAppDispatch();
+
+  const [isError, setIsError] = useState(false);
+  const [isThankYouPage, setIsThankYouPage] = useState(false);
+
+  const {
+    isApplyLoading,
+    retentionActions: {
+      extensionDetails: { periodUnit, amount },
+      offerId
+    }
+  } = useAppSelector(selectRetentionActions);
+
+  const handleApplyRetentionAction = async () => {
+    await dispatch(fetchApplyRetentionAction(offerId))
+      .unwrap()
+      .catch(() => {
+        setIsError(true);
+      });
+
+    setIsThankYouPage(true);
+  };
+
+  const renderPeriodTextElement = () => {
+    if (amount > 1) {
+      return (
+        <>
+          {`${amount} ${t(
+            `free-extension.period.${periodUnit}s`,
+            '{{periodUnit}}s',
+            {
+              periodUnit
+            }
+          )}`}
+        </>
+      );
+    }
+
+    return (
+      <>
+        {`${amount} ${t(
+          `free-extension.period.${periodUnit}`,
+          '{{periodUnit}}',
+          {
+            periodUnit
+          }
+        )}`}
+      </>
+    );
+  };
+
+  if (isError) {
+    return (
```

Comment:
maybe, we could create small presentional component like FreeExtensionError, FreeExtensionThankYou etc

Reply:
Ok, will do it